public/components/metrics/redux/slices/metrics_slice.ts

```diff
@@ -56,14 +56,16 @@ const fetchRemoteMetrics = async (pplService: any) => {
       pplService,
       `source = ${dataSource.DATASOURCE_NAME}.information_schema.tables`
     );
-    const normalizedData = catalogData.jsonData.map((obj: any) => ({
-      id: `${obj.TABLE_CATALOG}.${obj.TABLE_NAME}`,
-      name: `${obj.TABLE_CATALOG}.${obj.TABLE_NAME}`,
-      catalog: `${dataSource.DATASOURCE_NAME}`,
-      type: obj.TABLE_TYPE,
-      recentlyCreated: false,
-    }));
-    dataSet.push(normalizedData);
+    if (catalogData !== undefined) {
+      const normalizedData = catalogData.jsonData.map((obj: any) => ({
```

Comment:
Could you add type here for obj? and also a more semantic naming other than obj here would help more on understanding the code later. Also what is normalizedData?

Reply:
This is existing Sean's code, i just added if condition to fix a bug. I can change it if its immediately required